### PR TITLE
Use dom and new_inherited in (extendable)messageevent

### LINKED ISSUE #24707: "Messageport Gc'ed too early" panic in /fetch/security/embedded-credentials.tentative.sub.html
```
  ▶ CRASH [expected TIMEOUT] /fetch/security/embedded-credentials.tentative.sub.html
  │ 
  │ 
  │ 
  │ 
  │ [2019-11-11T14:02:51Z ERROR servo::browser] Multiple top level browsing contexts not supported yet.
  │ [2019-11-11T14:02:51Z ERROR servo::browser] Multiple top level browsing contexts not supported yet.
  │ [2019-11-11T14:02:51Z ERROR servo::browser] Multiple top level browsing contexts not supported yet.
  │ [2019-11-11T14:02:51Z ERROR servo::browser] Multiple top level browsing contexts not supported yet.
  │ Messageport Gc'ed too early (thread ScriptThread PipelineId { namespace_id: PipelineNamespaceId(1), index: PipelineIndex(1) }, at components/script/dom/globalscope.rs:513)
  │ stack backtrace:
  │    0: servo::main::{{closure}}
  │    1: std::panicking::rust_panic_with_hook
  │              at src/libstd/panicking.rs:472
  │    2: std::panicking::begin_panic
  │    3: script::dom::globalscope::GlobalScope::route_task_to_port
  │    4: <script::dom::globalscope::GlobalScope::start_message_port::process_pending_port_messages<F> as script::task::TaskOnce>::run_once
  │    5: <T as script::task::TaskBox>::run_box
  │    6: script::script_thread::ScriptThread::handle_msg_from_script
  │    7: _ZN6script13script_thread12ScriptThread11handle_msgs17h79a21672632590f6E.llvm.11879425879184375510
  │    8: script::script_thread::ScriptThread::start
  │    9: profile_traits::mem::ProfilerChan::run_with_memory_reporting
  │   10: std::sys_common::backtrace::__rust_begin_short_backtrace
  │   11: _ZN3std9panicking3try7do_call17h1ed362cc121fa29bE.llvm.11737484828325240857
  │   12: __rust_maybe_catch_panic
  │              at src/libpanic_unwind/lib.rs:80
  │   13: core::ops::function::FnOnce::call_once{{vtable.shim}}
  │   14: <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once
  │              at /rustc/246be7e1a557b8ac8287c6842379a0db67770be6/src/liballoc/boxed.rs:942
  │   15: <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once
  │              at /rustc/246be7e1a557b8ac8287c6842379a0db67770be6/src/liballoc/boxed.rs:942
  │       std::sys_common::thread::start_thread
  │              at src/libstd/sys_common/thread.rs:13
  │       std::sys::unix::thread::Thread::new::thread_start
  │              at src/libstd/sys/unix/thread.rs:79
  │   16: start_thread
  │   17: clone
  │ 
  │ [2019-11-11T14:02:52Z ERROR servo] Messageport Gc'ed too early
  └ Pipeline failed in hard-fail mode.  Crashing!
```
cc @gterzian 
Is this fixed by #24664?

### LINKED ISSUE #25464: MessageEvent and ExtendableMessageEvent have some issues
The implementations of `MessageEvent` and `ExtendableMessageEvent` have some issues.

Firstly, they store `Vec<DomRoot<T>>` in on their DOM object, they should be using `Vec<Dom<T>>`:

https://github.com/servo/servo/blob/a77318b424a2ea77198e3e0e6c5347b8c3092cc3/components/script/dom/messageevent.rs#L63

https://github.com/servo/servo/blob/a77318b424a2ea77198e3e0e6c5347b8c3092cc3/components/script/dom/extendablemessageevent.rs#L34

Secondly, they're constructing the type directly in `new()` instead of using a `new_inherited`

https://github.com/servo/servo/blob/a77318b424a2ea77198e3e0e6c5347b8c3092cc3/components/script/dom/extendablemessageevent.rs#L48-L54

https://github.com/servo/servo/blob/a77318b424a2ea77198e3e0e6c5347b8c3092cc3/components/script/dom/messageevent.rs#L86-L93

We should fix these issues.

cc @gterzian

### PR DESCRIPTION
**Pull request by gterzian**
Also FIX #24707

---

- [ ] `./mach build -d` does not report any errors
- [ ] `./mach test-tidy` does not report any errors
- [ ] These changes fix #25464 (GitHub issue number if applicable)

- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because ___

### CODE REVIEW

**Comment by gterzian** [comment]
@Manishearth r?

File: components/script/dom/messageevent.rs [hunk 1]
```diff
@@ -219,7 +238,13 @@ impl MessageEventMethods for MessageEvent {
...
     /// <https://html.spec.whatwg.org/multipage/#dom-messageevent-ports>
     fn Ports(&self, cx: JSContext) -> JSVal {
-        message_ports_to_frozen_array(self.ports.borrow().as_slice(), cx)
+        let ports: Vec<DomRoot<MessagePort>> = self
+            .ports
+            .borrow()
+            .iter()
+            .map(|port| DomRoot::from_ref(&**port))
+            .collect();
+        message_ports_to_frozen_array(ports.as_slice(), cx)
```
Comment (Manishearth) [inline components/script/dom/messageevent.rs]: We should update `message_ports_to_frozen_array` to work on `&[T] where T: ToJSValConvertible`  instead, and once we do that we should make it generic. But perhaps not in this PR.

**Comment by Manishearth** [comment]
This is r=me, but I'm going to wait for https://github.com/servo/servo/pull/25463 to land because Taskcluster has that build cached

**Comment by Manishearth** [comment]
@bors-servo r- retry

forgot that bors remembers results

**Comment by gterzian** [comment]
This seems to have also positively affected https://github.com/servo/servo/issues/24707 (which doesn't crash anymore), and https://github.com/servo/servo/issues/24810 (which now has a different error).
 cc @jdm

**Pull request merged**